tell_me_your_secrets/__main__.py

```diff
@@ -53,47 +53,47 @@
 
 
 class Signature(metaclass=abc.ABCMeta):
-    def __init__(self,part,name,signature):
+    def __init__(self, part: str, name: str, signature: str):
         self.part = part
         self.name = name
         self.signature = signature
     
     @abc.abstractmethod
-    def match(self, file_path,file_content) -> bool:
+    def match(self, file_path: str, file_content: str) -> bool:
         """Match Input of the With Signature of the part and Signature and Type of matching."""
         raise NotImplemented
 
+    def __str__(self):
+        return f'Type:{self.__class__} Name:{self.name} Part:{self.part}: Signature:{self.signature}'
+
 
 class RegexSignature(Signature):
-    def __init__(self, part, name,signature):
-        super().__init__(part, name,signature)
+    def __init__(self, part: str, name: str, signature: str):
+        super().__init__(part, name, signature)
         self.regex = re.compile(self.signature)
           
-    def match(self, file_path,file_content) -> bool:
+    def match(self, file_path: str, file_content: str) -> bool:
         compare_variable = None
         if self.part == 'extension':
-            compare_variable = find_extension(file_path)
+            compare_variable = find_extension(file_path)[1:]
```

Comment:
This was actually returning like `.txt` when we the base rules never define leading period

Reply:
Fascinating. Is it happening for all extensions?

Reply:
You can try it yourself

```
malachi@pulsar-work ~ $ python
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
>>> pathlib.Path('/home/malachi/Downloads/output.log').suffix
'.log'
```

We could add additional test cases around https://github.com/valayDave/tell-me-your-secrets/pull/8/files#diff-e5d9203764fd4dc92a48e417f9f961bdR20

But would require some IO mocking I suspect

Reply:
@valayDave the alternative would be to suffix all the regex `extension` rules with `\.`

Reply:
This should be fine.